src/Functions/custom/New-AzFunctionApp.ps1

```diff
@@ -0,0 +1,364 @@
+        [System.String]
+        ${StorageAccountName},        
+
+        [Parameter(ParameterSetName="Consumption", HelpMessage='Name of the existing App Insights project to be added to the function app.')]
+        [Parameter(ParameterSetName="ByAppServicePlan")]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        [Alias("AppInsightsName")]
+        ${ApplicationInsightsName},
+
+        [Parameter(ParameterSetName="Consumption", HelpMessage='Instrumentation key of App Insights to be added.')]
+        [Parameter(ParameterSetName="ByAppServicePlan")]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        [Alias("AppInsightsKey")]
+        ${ApplicationInsightsKey},
+
+        [Parameter(ParameterSetName="Consumption", HelpMessage='The location for the consumption plan.')]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        ${Location},
+
+        [Parameter(ParameterSetName="ByAppServicePlan", HelpMessage='The name of the service plan.')]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        ${PlanName},
+
+        [Parameter(ParameterSetName="ByAppServicePlan", HelpMessage='The OS to host the function app.')]
+        [Parameter(ParameterSetName="Consumption")]
+        [ArgumentCompleter([Microsoft.Azure.PowerShell.Cmdlets.Functions.Support.WorkerType])]
+        # OS type (Linux or Windows)
+        ${OSType},
+        
+        [Parameter(ParameterSetName="ByAppServicePlan", HelpMessage='The function runtime.')]
+        [Parameter(ParameterSetName="Consumption")]
+        [ArgumentCompleter([Microsoft.Azure.PowerShell.Cmdlets.Functions.Support.RuntimeType])]
+        # Runtime type (DotNet, Node, Java, PowerShell or Python
+        [System.String]
+        ${Runtime},
+
+        [Parameter(ParameterSetName="ByAppServicePlan", HelpMessage='Disable creating application insights resource during the function app creation. No logs will be available.')]
+        [Parameter(ParameterSetName="Consumption")]
+        [System.Management.Automation.SwitchParameter]
+        [Alias("DisableAppInsights")]
+        ${DisableApplicationInsights},
+        
+        [Parameter(ParameterSetName="ByAppServicePlan")]
+        [Parameter(ParameterSetName="Consumption", HelpMessage='Returns true when the command succeeds.')]
+        [System.Management.Automation.SwitchParameter]
+        ${PassThru},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        ${NoWait},
+
+        [Parameter()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        ${AsJob},
+
+        [Alias('AzureRMContext', 'AzureCredential')]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Azure')]
+        [System.Management.Automation.PSObject]
+        ${DefaultProfile}
+)
```

Comment:
Same comment on the DontShow parameters - this applies across all the custom cmdlets

Reply:
Here is the list of parameters I have added to all cmdlets. Please let me know if I missed anything. 

```
        [Parameter(DontShow)]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [System.Management.Automation.SwitchParameter]
        # Wait for .NET debugger to attach
        ${Break},

        [Parameter(DontShow)]
        [ValidateNotNull()]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Runtime.SendAsyncStep[]]
        # SendAsync Pipeline Steps to be appended to the front of the pipeline
        ${HttpPipelineAppend},

        [Parameter(DontShow)]
        [ValidateNotNull()]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Runtime.SendAsyncStep[]]
        # SendAsync Pipeline Steps to be prepended to the front of the pipeline
        ${HttpPipelinePrepend},

        [Parameter(DontShow)]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [System.Uri]
        # The URI for the proxy server to use
        ${Proxy},

        [Parameter(DontShow)]
        [ValidateNotNull()]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [System.Management.Automation.PSCredential]
        # Credentials for a proxy server to use for the remote call
        ${ProxyCredential},

        [Parameter(DontShow)]
        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
        [System.Management.Automation.SwitchParameter]
        # Use the default credentials for the proxy
        ${ProxyUseDefaultCredentials}
```